### ISSUE TITLE
Synth: param_iterator

### ISSUE THREAD

**Issue by TomDonoghue** [voytekresearch/fooof#96]
Add a new function to fooof.synth for iterating across a range of parameters to be explored.

**Comment by TomDonoghue** [comment]
Note that this is WIP on the 'paramiter' branch. 

To do list:
- [ ] Update docs, to be proper numpy doc format
- [ ] Add tests under fooof/tests/test_synth.py
- [ ] Think about, and set default (probably no defaults).
- [ ] Sanity check / think about the API

Discussion points:
- [ ] How do we deal with using this with multiple oscillations
- [ ] Currently, only ever iterate one thing. Stay with this?

**Comment by TomDonoghue** [comment]
Proposal of how to extend the param_iter to have multiple oscillation definitions, etc:


Iterators should be defined, by the user, outside param_iter (with number_iter).
Then param_iter just has to find where iterators are defined, to yield them.
This extends to defining arbitrality many oscillations, and could extend to many iterators

```
# Example, definition
osc_1 = [number_iter(8, 12, 0.1), 1, 1] # Alpha osc, stepping across CF
osc_2 = [20, 2, 2]

oscillation_params = [osc_1, osc_2]
osc_iterator = param_iter(oscillation_params)
```

**Comment by TomDonoghue** [comment]
This was merged in #99
